### ISSUE TITLE
XML_Toolkit: Fix Normal direction

### ISSUE THREAD

**Issue by michaldengusiak** [BHoM/XML_Toolkit#166]
We need to revert normals - point direction for panels...
![image](https://user-images.githubusercontent.com/26113670/51108190-35445400-17e9-11e9-955f-30099b0b919b.png)

grey should in inside and color outside 
![image](https://user-images.githubusercontent.com/26113670/51108296-9e2bcc00-17e9-11e9-9d23-7d117c1193f1.png)

**Comment by rwemay** [comment]
May be worth @IsakNaslundBh sharing how we're intending to store surface normals on panelplanar elements. @pawelbaran FYI

**Comment by IsakNaslundBh** [comment]
Think we have two options for the normals. 

1. Add a boolean, together with a convention, like if the normal is non-vertical, true means an upwards normal, if vertical, tries to align with Global Y/x.
1. Based on the edge order of the defining outline. Something like, the edges always run counterclockwise around the normal, which then makes it possible to back calc the normal direction based on the outline.

Think a lot of geometry libraries goes with something like the second option, but could be a bit more fiddly to change etc.

**Comment by rwemay** [comment]
I wouldn’t necessarily limit it to normal actually - I can see a use case for coordinate system, similar to str panel. Of the top of my head - polarising screens (with anisotropic light properties).

**Comment by IsakNaslundBh** [comment]
yes! Then I think the idea would be to simply have an angle double + a convention for for planar elements.

Then the coordinate system could be calculated for non-vertical panels as for example, if we don't find a better convention:

Z = normal based on previous comment
X = Global X projected on the normal plane, then rotated angle radians counter clockwise
Y = Z.CrossProduct(X)

For vertical panels we will need to alter the convention, but still doable. Will work very similar to the orientation angle for bars and ensure no duplicate information. ETABS solves orientation of panels in this manner (not 100% sure on the exact convention, but can look it up).

**Comment by FraserGreenroyd** [comment]
Alternative theory - I'm trying to decorate the inside to make it look prettier than a grey interior? I mean imagine living in grey walls! Just awful!

I'll take a look in the next upgrade sprint for XML at the above suggestions :smile:

**Comment by al-fisher** [comment]
> prettier than a grey interior? 

Bright Magenta?! Crumbs :sunglasses:  
:laughing:

**Comment by FraserGreenroyd** [comment]
![image](https://user-images.githubusercontent.com/18049174/51187189-2c2fb180-18d3-11e9-974a-be3485ec435d.png)

There you go, is that better? Painting the outside like loons... I dunno... This won't win any prizes on Grand Designs I'll tell you that!
